test/e2e_env/kubernetes/meshmetric/meshmetric.go

```diff
@@ -0,0 +1,249 @@
+          port: 8080
+          path: /metrics
+          tls:
+            mode: Disabled
+`, policyName, Config.KumaNamespace, mesh)
+	return YamlK8s(meshMetric)
+}
+
+func MeshMetricWithSpecificPrometheusBackend(policyName string, mesh string, clientId string) InstallFunc {
+	meshMetric := fmt.Sprintf(`
+apiVersion: kuma.io/v1alpha1
+kind: MeshMetric
+metadata:
+  name: %s
+  namespace: %s
+  labels:
+    kuma.io/mesh: %s
+spec:
+  targetRef:
+    kind: Mesh
+  default:
+    backends:
+      - type: Prometheus
+        prometheus: 
+          clientId: %s
+          port: 8080
+          path: /metrics
+          tls:
+            mode: Disabled
+`, policyName, Config.KumaNamespace, mesh, clientId)
+	return YamlK8s(meshMetric)
+}
+
+func MeshMetricWithSpecificPrometheusBackendForMeshService(mesh string, clientId string, serviceName string) InstallFunc {
+	meshMetric := fmt.Sprintf(`
+apiVersion: kuma.io/v1alpha1
+kind: MeshMetric
+metadata:
+  name: mesh-metric-2
+  namespace: %s
+  labels:
+    kuma.io/mesh: %s
+spec:
+  targetRef:
+    kind: MeshService
+    name: %s
+  default:
+    backends:
+      - type: Prometheus
+        prometheus: 
+          clientId: %s
+          port: 8080
+          path: /metrics
+          tls:
+            mode: Disabled
+`, Config.KumaNamespace, mesh, serviceName, clientId)
+	return YamlK8s(meshMetric)
+}
+
+func MeshMetric() {
+	const (
+		namespace             = "meshmetric"
+		mainMesh              = "main-metrics-mesh"
+		mainPrometheusId      = "main-prometheus"
+		secondaryMesh         = "secondary-metrics-mesh"
+		secondaryPrometheusId = "secondary-prometheus"
+	)
+
+	BeforeAll(func() {
+		err := NewClusterSetup().
+			Install(MeshKubernetes(mainMesh)).
+			Install(MeshKubernetes(secondaryMesh)).
+			Install(NamespaceWithSidecarInjection(namespace)).
+			Setup(kubernetes.Cluster)
+		Expect(err).To(Succeed())
+
+		for i := 0; i < 3; i++ {
+			Expect(
+				kubernetes.Cluster.Install(testserver.Install(
+					testserver.WithName(fmt.Sprintf("test-server-%d", i)),
+					testserver.WithMesh(mainMesh),
+					testserver.WithNamespace(namespace),
+				)),
+			).To(Succeed())
+		}
+		for i := 3; i < 6; i++ {
+			Expect(
+				kubernetes.Cluster.Install(testserver.Install(
+					testserver.WithName(fmt.Sprintf("test-server-%d", i)),
+					testserver.WithMesh(secondaryMesh),
+					testserver.WithNamespace(namespace),
+				)),
+			).To(Succeed())
+		}
+	})
+
+	E2EAfterEach(func() {
+		Expect(DeleteMeshResources(kubernetes.Cluster, mainMesh, v1alpha1.MeshMetricResourceTypeDescriptor)).To(Succeed())
+		Expect(DeleteMeshResources(kubernetes.Cluster, secondaryMesh, v1alpha1.MeshMetricResourceTypeDescriptor)).To(Succeed())
+	})
+
+	E2EAfterAll(func() {
+		Expect(kubernetes.Cluster.TriggerDeleteNamespace(namespace)).To(Succeed())
+		Expect(kubernetes.Cluster.DeleteMesh(mainMesh)).To(Succeed())
+		Expect(kubernetes.Cluster.DeleteMesh(secondaryMesh)).To(Succeed())
+	})
+
+	It("Basic MeshMetric", func() {
```

Comment:
maybe something like:

```suggestion
	It("Basic MeshMetric policy exposes Envoy metrics on correct port", func() {
```

?

Reply:
changed